ksql-engine/src/test/java/io/confluent/ksql/connect/ConnectConfigServiceTest.java

```diff
@@ -101,13 +101,15 @@ public void shouldCreateConnectorFromConfig() throws InterruptedException {
   }
 
   @Test
-  public void shouldWakeupConsumerBeforeShuttingDown() {
+  public void shouldWakeupConsumerBeforeShuttingDown() throws InterruptedException {
     // Given:
+    final CountDownLatch noMoreLatch = new CountDownLatch(1);
     setupConfigService();
-    givenNoMoreRecords(when(consumer.poll(any())));
+    givenNoMoreRecords(when(consumer.poll(any())), noMoreLatch);
     configService.startAsync().awaitRunning();
```

Comment:
Minor: Isn't this where the action happens? If so, would this be better suited for "when"?

Reply:
I suppose so! I thought that the "action" happens when `stopAsync()` is called (e.g. it calls `triggerShutdown` which in turn causes the wakeup/close, but I think it makes sense to put it in "when' as well. Will make that change.